proc/proc.go

```diff
@@ -109,48 +112,111 @@ func GetKernelModules(modulesPath string,
 		Size:    int(etext.Address - stext.Address),
 	})
 
-	atLeastOneValidAddress := false
-	count := 0
+	content, err := io.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read modules file: %v", err)
+	}
 
-	var scanner = bufio.NewScanner(file)
-	for scanner.Scan() {
-		var size, address uint64
-		var refcount int64
-		var name, dependencies, state string
+	modules, err := parseKernelModules(content)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel modules: %v", err)
+	}
+
+	for _, kmod := range modules {
+		symmap.Add(libpf.Symbol{
+			Name:    libpf.SymbolName(kmod.name),
+			Address: libpf.SymbolValue(kmod.address),
+			Size:    kmod.size,
+		})
+	}
+
+	symmap.Finalize()
 
+	return &symmap, nil
+}
+
+func parseKernelModules(content []byte) ([]kernelModule, error) {
+	var (
+		modules                []kernelModule
+		scanner                = bufio.NewScanner(bytes.NewReader(content))
```

Comment:
Why slurp all the content and then construct a scanner on top of it -- why not construct a scanner directly on top of the file like we did in the old code?

Doesn't really matter since these files should be small but it seems semantically unnecessary.